Sources/SwiftCompilerPluginMessageHandling/PluginMacroExpansionContext.swift

```diff
@@ -170,19 +170,35 @@ class SourceManager {
     from startKind: PositionInSyntaxNode = .afterLeadingTrivia,
     to endKind: PositionInSyntaxNode = .beforeTrailingTrivia
   ) -> SourceRange? {
-    guard let base = self.knownSourceSyntax[node.root.id] else {
+    if let base = self.knownSourceSyntax[node.root.id] {
+      return range(node.position(at: startKind)..<node.position(at: endKind), in: base)
+    } else {
       return nil
     }
-    let localStartPosition = node.position(at: startKind)
-    let localEndPosition = node.position(at: endKind)
-    precondition(localStartPosition <= localEndPosition)
```

Comment:
Should we keep this precondition?

Reply:
Hasn't this precondition been enforced by the range constructor?

Reply:
True